proposals/csharp-7.2/ref-local-reassignment.md

```diff
@@ -0,0 +1,27 @@
+# Ref Local Reassignment
+
+In C# 7.2, we add support for rebinding the referent of a ref local variable or a ref parameter.
```

Comment:
There is nothing technically wrong with rebinding ref parameters. I guess - why not. Similarly as we allow assigning to byval params.

Reply:
Could it cause confusion? I assume it would be detaching from the passed ref; but it may be assumed that its changing the passed ref (in the callee) rather than just the local parameter ref? 

Unless it actually is ofc... :)

Reply:
Caller will see no effects from/after param rebinding by callee.  But that could be the reason why callee does this...
 It will not be the most common feature to use - like assignments to ordinary parameters, but nothing is wrong with it technically.

Reply:
Maybe I'm just over thinking it :) 

People don't think changing a non-ref param changes it in callee and its not a `ref ref` or `&ref` type.

Also is probably handy so you have a starter `ref` that you then change rather than having to introduce a local copy that you then modify.

e.g.
```csharp
ref T FindFrom(ref T current)
{
    ref var local = ref current;
    while (!local.IsCorrect)
    {
        local = ref _array[local.Next];
    }
    
    return ref local;
}
```
would be better as
```csharp
ref T FindFrom(ref T current)
{
    while (!current.IsCorrect)
    {
        current = ref _array[current.Next];
    }
    
    return ref current;
}
```

Reply:
Though in practice it would be more like
```csharp
static T NotFound;

T[] _array;

ref T FindFrom(ref T current, out bool success)
{
    var maxSteps = _array.Length;
    success = true;
    do
    {
        if (current.IsCorrect)
        {
            return ref current;
        }
        else if (current.Next < 0)
        {
            break;
        }
        current = ref _array[current.Next];
        maxSteps--;
    } while (maxSteps > 0);

    success = false;
    return ref NotFound;
}
```

Reply:
Also need to clarify position on rebinding of ‘this’ in struct methods. In theory all the arguments for allowing rebinding of ref parameters apply to ‘this’ in a struct.  
So is the ‘this = ref something’ allowed in a struct method?

Reply:
Just as "this" in a class acts like a parameter, it cannot be assigned, "this" in a struct acts like a ref parameter, but cannot be rebound. I'll add that to the spec.